src/bin/ion/commands/inspect.rs

```diff
@@ -187,22 +187,39 @@ impl<'a, 'b> IonInspector<'a, 'b> {
         Ok(inspector)
     }
 
+    fn confirm_encoding_is_supported(&self, encoding: IonEncoding) -> Result<()> {
+        use IonEncoding::*;
+        match encoding {
+            Text_1_0 | Text_1_1 => {
+                bail!("`inspect` does not support text Ion streams.");
+            }
+            Binary_1_0 => Ok(()),
+            Binary_1_1 => bail!("`inspect` does not yet support binary Ion v1.1"),
+            // `IonEncoding` is #[non_exhaustive]
+            _ => bail!("`inspect does not yet support {}", encoding.name()),
+        }
+    }
```

Comment:
🗺️ The latest `ion_rs` makes it possible to get the `IonEncoding` from any raw stream item. This allows us to trivially confirm that we're working with what we expect; in `inspect`'s case, that's binary Ion 1.0.